demos/http/http_demo_s3_upload/http_demo_s3_upload.c

```diff
@@ -0,0 +1,626 @@
+#include "http_demo_utils.h"
+
+/* HTTP API header. */
+#include "core_http_client.h"
+
+/* OpenSSL transport header. */
+#include "openssl_posix.h"
+
+/* Retry utilities. */
+#include "retry_utils.h"
+
+/* Check that TLS port of the server is defined. */
+#ifndef HTTPS_PORT
+    #error "Please define a HTTPS_PORT."
+#endif
+
+/* Check that a path for Root CA Certificate is defined. */
+#ifndef ROOT_CA_CERT_PATH
+    #error "Please define a ROOT_CA_CERT_PATH."
+#endif
+
+/* Check that the pre-signed PUT URL is defined. */
+#ifndef S3_PRESIGNED_PUT_URL
+    #error "Please define a S3_PRESIGNED_PUT_URL."
+#endif
+
+/* Check that the pre-signed GET URL is defined. */
+#ifndef S3_PRESIGNED_GET_URL
+    #error "Please define a S3_PRESIGNED_GET_URL."
+#endif
+
+
+/**
+ * @brief ALPN protocol name to be sent as part of the ClientHello message.
+ *
+ * @note When using ALPN, port 443 must be used to connect to AWS IoT Core.
+ */
+#define IOT_CORE_ALPN_PROTOCOL_NAME               "\x0ex-amzn-http-ca"
+
+/**
+ * @brief Field name of the HTTP Range header to read from server response.
+ */
+#define HTTP_CONTENT_RANGE_HEADER_FIELD           "Content-Range"
+
+/**
+ * @brief Length of the HTTP Range header field.
+ */
+#define HTTP_CONTENT_RANGE_HEADER_FIELD_LENGTH    ( sizeof( HTTP_CONTENT_RANGE_HEADER_FIELD ) - 1 )
+
+/**
+ * @brief Delay in seconds between each iteration of the demo.
+ */
+#define DEMO_LOOP_DELAY_SECONDS                   ( 5U )
+
+/* Check that transport timeout for transport send and receive is defined. */
+#ifndef TRANSPORT_SEND_RECV_TIMEOUT_MS
+    #define TRANSPORT_SEND_RECV_TIMEOUT_MS    ( 1000 )
+#endif
+
+/* Check that size of the user buffer is defined. */
+#ifndef USER_BUFFER_LENGTH
+    #define USER_BUFFER_LENGTH    ( 2048 )
+#endif
+
+/* Pointer to the data to upload.*/
+#ifndef DEMO_HTTP_UPLOAD_DATA
+    #define DEMO_HTTP_UPLOAD_DATA    "Hello World!"
+#endif
+
+/* The size of the data in bytes to upload. */
+#ifndef DEMO_HTTP_UPLOAD_DATA_SIZE
+    #define DEMO_HTTP_UPLOAD_DATA_SIZE    ( sizeof( DEMO_HTTP_UPLOAD_DATA ) - 1 )
+#endif
+
+/**
+ * @brief A buffer used in the demo for storing HTTP request headers and
+ * HTTP response headers and body.
+ *
+ * @note This demo shows how the same buffer can be re-used for storing the HTTP
+ * response after the HTTP request is sent out. However, the user can also
+ * decide to use separate buffers for storing the HTTP request and response.
+ */
+static uint8_t userBuffer[ USER_BUFFER_LENGTH ];
+
+/**
+ * @brief Represents header data that will be sent in an HTTP request.
+ */
+static HTTPRequestHeaders_t requestHeaders;
+
+/**
+ * @brief Configurations of the initial request headers that are passed to
+ * #HTTPClient_InitializeRequestHeaders.
+ */
+static HTTPRequestInfo_t requestInfo;
+
+/**
+ * @brief Represents a response returned from an HTTP server.
+ */
+static HTTPResponse_t response;
+
+/* The host address string extracted from the pre-signed URL. */
+static char serverHost;
```

Comment:
Missing @brief tag

Reply:
Also, shouldn't this be a `char *`?

Reply:
It was supposed to be a char[] to extract the host name from the address pointer. Updated to fix that.